test/integration/server.cc

```diff
@@ -107,36 +100,59 @@ void IntegrationTestServer::onWorkerListenerRemoved() {
   }
 }
 
+void IntegrationTestServer::serverReady() {
+  pending_listeners_ = server().listenerManager().listeners().size();
+  server_set_.setReady();
+}
+
 void IntegrationTestServer::threadRoutine(const Network::Address::IpVersion version,
                                           bool deterministic) {
   OptionsImpl options(Server::createTestOptionsImpl(config_path_, "", version));
-  Server::HotRestartNopImpl restarter;
   Thread::MutexBasicLockable lock;
 
-  ThreadLocal::InstanceImpl tls;
-  Stats::HeapStatDataAllocator stats_allocator;
-  Stats::StatsOptionsImpl stats_options;
-  Stats::ThreadLocalStoreImpl stats_store(stats_options, stats_allocator);
-  stat_store_ = &stats_store;
   Runtime::RandomGeneratorPtr random_generator;
   if (deterministic) {
     random_generator = std::make_unique<testing::NiceMock<Runtime::MockRandomGenerator>>();
   } else {
     random_generator = std::make_unique<Runtime::RandomGeneratorImpl>();
   }
-  server_.reset(new Server::InstanceImpl(
-      options, time_system_, Network::Utility::getLocalAddress(version), *this, restarter,
-      stats_store, lock, *this, std::move(random_generator), tls));
-  pending_listeners_ = server_->listenerManager().listeners().size();
-  ENVOY_LOG(info, "waiting for {} test server listeners", pending_listeners_);
-  // This is technically thread unsafe (assigning to a shared_ptr accessed
-  // across threads), but because we synchronize below on server_set, the only
-  // consumer on the main test thread in ~IntegrationTestServer will not race.
```

Comment:
Is this comment no longer relevant? If so, why?

Reply:
Thanks for catching that; it's still relevant.  Re-added.